src/sqlancer/common/query/SQLQueryAdapter.java

```diff
@@ -46,7 +50,7 @@ private String canonicalizeString(String s) {
     }
 
     private void checkQueryString() {
-        if (query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN") && !couldAffectSchema) {
+        if (guessAffectSchemaFromQuery(query) != couldAffectSchema) {
```

Comment:
The comparison here confuses me. I think the pattern before was more readable?

Reply:
I move the `query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN")` part to a separate method, for guess `couldAffectSchema` when not given, and to check here.

Reply:
I think extracting this part is fine, but the "!=" is confusing.

Reply:
I think only when the flag `couldAffectSchema` is false, we need to check whether such value is proper according to the result of function `guessAffectSchemaFromQuery`. In this case, an exception could occur, such as when setting `couldAffectSchema` to true for ALTER statement. 
The following revision may be acceptable.
```java
    private void checkQueryString() {
        if (!couldAffectSchema) {
            if (guessAffectSchemaFromQuery(query)) {
                throw new AssertionError("CREATE TABLE statements should set couldAffectSchema to true");
            }
        }
    }
```

Reply:
Yes, I think that would be easy to understand, so I would prefer this.

Reply:
Great, I have pushed the changes.
PTAL